pkg/daemon/drain.go

```diff
@@ -248,50 +248,53 @@ func isSafeContainerRegistryConfChanges(oldIgnConfig, newIgnConfig ign3types.Con
 	// Check for modified registry
 	for regLoc, newReg := range newRegHashMap {
 		oldReg, ok := oldRegHashMap[regLoc]
-		if ok && !reflect.DeepEqual(oldReg, newReg) {
-			// Registry is available in both old and new config and some changes has been found.
-			// Check that changes made are safe or not.
-			if oldReg.Prefix != newReg.Prefix {
-				glog.Infof("%s: prefix value for registry %s has changed from %s to %s",
-					constants.ContainerRegistryConfPath, regLoc, oldReg.Prefix, newReg.Prefix)
-				return false, nil
-			}
-			if oldReg.Location != newReg.Location {
-				glog.Infof("%s: location value for registry %s has changed from %s to %s",
-					constants.ContainerRegistryConfPath, regLoc, oldReg.Location, newReg.Location)
-				return false, nil
-			}
-			if oldReg.Blocked != newReg.Blocked {
-				glog.Infof("%s: blocked value for registry %s has changed from %t to %t",
-					constants.ContainerRegistryConfPath, regLoc, oldReg.Blocked, newReg.Blocked)
-				return false, nil
-			}
-			if oldReg.Insecure != newReg.Insecure {
-				glog.Infof("%s: insecure value for registry %s has changed from %t to %t",
-					constants.ContainerRegistryConfPath, regLoc, oldReg.Insecure, newReg.Insecure)
-				return false, nil
-			}
-			if oldReg.MirrorByDigestOnly == newReg.MirrorByDigestOnly {
-				// Ensure that all the old mirrors are present
-				for _, m := range oldReg.Mirrors {
-					if !searchRegistryMirror(m.Location, newReg.Mirrors) {
-						glog.Infof("%s: mirror %s has been removed in registry %s",
-							constants.ContainerRegistryConfPath, m.Location, regLoc)
-						return false, nil
-					}
+		if ok {
+			// Registry is available in both old and new config.
+			if !reflect.DeepEqual(oldReg, newReg) {
```

Comment:
Took this if-statement that checks whether `oldReg` and `newReg` are equal inside the `ok` if block, so that if an existing/old registry with `mirror-by-digest-only=false` hasn't been changed, it will not cause a reboot. Thoughts @sinnykumari?

Reply:
Good catch, indeed comparing registry logic should be a separate block.